src/gui/addnewtorrentdialog.cpp

```diff
@@ -91,15 +92,36 @@ AddNewTorrentDialog::AddNewTorrentDialog(QWidget *parent)
 
     auto session = BitTorrent::Session::instance();
 
-    ui->startTorrentCheckBox->setChecked(!session->isAddTorrentPaused());
+    // TODO: set dialog file properties using m_torrentParams.filePriorities 
+    
+    if (m_torrentParams.addPaused == TriStateBool::True) {
+        ui->startTorrentCheckBox->setChecked(false);
+    }
+    else if (m_torrentParams.addPaused == TriStateBool::False) {
+        ui->startTorrentCheckBox->setChecked(true);
+    }
+    else {
+        ui->startTorrentCheckBox->setChecked(!session->isAddTorrentPaused());
+    }
+    
+    if (m_torrentParams.createSubfolder == TriStateBool::True) {
+        ui->createSubfolderCheckBox->setChecked(true);
+    }
+    else if (m_torrentParams.createSubfolder == TriStateBool::False) {
+        ui->createSubfolderCheckBox->setChecked(false);
+    }
+    else {
+        ui->createSubfolderCheckBox->setChecked(session->isCreateTorrentSubfolder());
+    }
+    
     ui->comboTTM->blockSignals(true); // the TreeView size isn't correct if the slot does it job at this point
     ui->comboTTM->setCurrentIndex(!session->isAutoTMMDisabledByDefault());
     ui->comboTTM->blockSignals(false);
     populateSavePathComboBox();
     connect(ui->savePathComboBox, SIGNAL(currentIndexChanged(int)), SLOT(onSavePathChanged(int)));
     connect(ui->browseButton, SIGNAL(clicked()), SLOT(browseButton_clicked()));
     ui->defaultSavePathCheckBox->setVisible(false); // Default path is selected by default
-    ui->createSubfolderCheckBox->setChecked(session->isCreateTorrentSubfolder());
+    ui->skipCheckingCheckBox->setChecked(m_torrentParams.skipChecking);
```

Comment:
Not technically a replacement. The code for setting `ui->createSubfolderCheckBox` is higher up in the function now so that it is grouped with similarly structured code for setting `ui->startTorrentCheckBox`, and `ui->skipCheckingCheckBox` is an addition now that `BitTorrent::AddTorrentParams` is a parameter the constructor.

I do agree with you that this would be much easier to follow if commits weren't being squashed. Let me know if you don't want me to do that any longer.